src/operations/update.ts

```diff
@@ -40,7 +40,7 @@ export interface UpdateResult {
   /** The number of documents that were upserted */
   upsertedCount: number;
   /** The identifier of the inserted document if an upsert took place */
-  upsertedId: ObjectId;
+  upsertedId: ObjectId | null;
```

Comment:
This doesn't account for [setOnInsert](https://www.mongodb.com/docs/v6.0/reference/operator/update/setOnInsert/#example)